doc/port_auto_neg/port-auto-negotiation-design.md

```diff
@@ -396,9 +396,23 @@ To support validate interface speed on CLI side, a new field **supported_speeds*
 	; field                 = value
     ...
 	supported_speeds        = STRING                         ; supported speed list
+  speed                   = STRING                         ; operational speed
 
 An example value of supported_speeds could be "10000,25000,40000,100000".
 
+Before this feature, port speed in APP DB indicates both the configured speed and the operational speed. It is OK without this feature because port operational speed must be configured speed or port operational status is down. However, this is not true with this feature. Consider following flow:
+
+1. Configure port speed to 100G
+2. Configure advertised speed to 50G and enable auto negotiation
+3. Port operational speed turns to 50G
+4. Configure any port attribute, e.g mtu, portsyncd would put all port attributes to APP DB
+5. `show interface status` displays port speed as 100G which is incorrect
+
+To overcome this issue, following changes are required:
+
+1. Put port operational speed to STATE DB PORT_TABLE
+2. intfutil, portstat, voqutil shall be change to get port operational speed from STATE DB first. For backward compatible, intfutil, portstat, voqutil shall still get port operational speed from APP DB if port speed is not available in APP DB.
```

Comment:
typo:   shall still get port operational speed from **APP DB** if port speed is not available in **APP DB**.